listenbrainz/webserver/views/atom.py

```diff
@@ -0,0 +1,570 @@
+        return dt.strftime("%Y")
+    elif stats_range in [StatisticsRange.quarter.value]:
+        return f"Q{int((dt.month - 1) / 3) + 1} {dt.year}"
+    elif stats_range in [StatisticsRange.half_yearly.value]:
+        return f"H{int((dt.month - 1) / 6) + 1} {dt.year}"
+    elif stats_range in [StatisticsRange.all_time.value]:
+        return "All Time"
+    return ""
+
+
+def _init_feed(id, title, self_url, alternate_url):
+    fg = FeedGenerator()
+    fg.id(id)
+    fg.title(title)
+    fg.author({"name": "ListenBrainz"})
+    fg.link(href=alternate_url, rel="alternate")
+    fg.link(href=self_url, rel="self")
+    fg.logo(_external_url_for("static", filename="img/listenbrainz_logo_icon.svg"))
+    fg.language("en")
+    return fg
+
+
+@atom_bp.route("/user/<user_name>/listens", methods=["GET"])
+@crossdomain
+@ratelimit()
+@api_listenstore_needed
+def get_listens(user_name):
+    """
+    Get listens feed for a user.
+
+    :param minutes: The time interval in minutes from current time to fetch listens for.
+                    For example, if minutes=60, listens from the last hour will be fetched. Default is 60.
+    :statuscode 200: The feed was successfully generated.
+    :statuscode 400: Bad request.
+    :statuscode 404: The user does not exist.
+    :resheader Content-Type: *application/atom+xml*
+    """
+    user = db_user.get_by_mb_id(db_conn, user_name)
+    if user is None:
+        return Response(status=404)
+
+    minutes = request.args.get("minutes", DEFAULT_MINUTES_OF_LISTENS)
+    if minutes:
+        try:
+            minutes = int(minutes)
+        except ValueError:
+            return Response(status=400)
+    if minutes < 1 or minutes > MAX_MINUTES_OF_LISTENS:
+        return Response(status=400)
+
+    # estimate limit from minutes, assuming 1 listen per minute.
+    # mostly likely there won't be that much listens, but with such padding
+    # it's less likely for feed readers to miss listens.
+    limit = minutes
+
+    to_ts = datetime.now()
+    listens, _, _ = timescale_connection._ts.fetch_listens(
+        user, to_ts=to_ts, limit=limit
+    )
+
+    fg = _init_feed(
+        _external_url_for(".get_listens", user_name=user_name),
+        f"Listens for {user_name} - ListenBrainz",
+        _external_url_for("user.index", path="", user_name=user_name),
+        _external_url_for(".get_listens", user_name=user_name),
+    )
+
+    # newer listen comes first
+    for listen in reversed(listens):
+        track_name = listen.data["track_name"]
+        recording_mbid = listen.data["additional_info"].get("recording_mbid")
+        artist_name = listen.data["artist_name"]
+        artist_mbid = (
+            listen.data["additional_info"].get("artist_mbids")[0]
+            if listen.data["additional_info"].get("artist_mbids")
+            else None
+        )
+
+        fe = fg.add_entry()
+        # according to spec, ids don't have to be deferencable.
```

Comment:
What do you mean with deferencable? Thats not an english word I know. :p

Reply:
In this case, it basically means the id string doesn't have to be a valid webpage URL, though I'm using HTTP scheme. It just needs to uniquelly identify a feed or feed entry.